src/backends/gpu/threshold_watcher.hpp

```diff
@@ -51,15 +48,15 @@ class threshold_watcher {
         thresholds_(memory::make_const_view(thresh)),
         prev_values_(values),
         is_crossed_(size()),
-        stack_(make_managed_ptr<stack_type>(10*size()))
+        stack_(10*size())
```

Comment:
Not in scope, but something for later: demystify the magic number `10`?

Reply:
Yeah, that is a hack. It deserves a comment to that effect.